src/TorSharp/TorSharpSettings.cs

```diff
@@ -40,25 +40,34 @@ public TorSharpSettings()
                 OSPlatform = TorSharpOSPlatform.Unknown;
             }
 
-            switch (RuntimeInformation.ProcessArchitecture)
+            Architecture = RuntimeInformation.ProcessArchitecture switch
             {
-                case SystemArchitecture.X86:
-                    Architecture = TorSharpArchitecture.X86;
-                    break;
-                case SystemArchitecture.X64:
-                    Architecture = TorSharpArchitecture.X64;
-                    break;
-                default:
-                    Architecture = TorSharpArchitecture.Unknown;
-                    break;
-            }
+                SystemArchitecture.X86 => TorSharpArchitecture.X86,
+                SystemArchitecture.X64 => TorSharpArchitecture.X64,
+                SystemArchitecture.Arm => TorSharpArchitecture.Arm32,
+                SystemArchitecture.Arm64 => TorSharpArchitecture.Arm64,
+                _ => TorSharpArchitecture.Unknown,
+            };
 #else
             OSPlatform = TorSharpOSPlatform.Windows;
             Architecture = Environment.Is64BitProcess ? TorSharpArchitecture.X64 : TorSharpArchitecture.X86;
 #endif
 
             PrivoxySettings = new TorSharpPrivoxySettings();
             TorSettings = new TorSharpTorSettings();
+
+#if NETSTANDARD
+            if (OSPlatform == TorSharpOSPlatform.Linux)
```

Comment:
This is a change in default behavior. Can we just scope this to the Arm32 and Arm64 architectures?

One drawback of defaulting to the system-provided on is that the Tor or Privoxy version on the system (i.e. provided by distro package maintainers) may not update to the latest version as quickly as what the tool fetcher in TorSharp can provide (which takes the latest build released by the upstream maintainers).

Reply:
It's a good point. Using the system version is probably better on Linux. 

If you want to change the defaults for non-ARM Linux, make an update to the README with an FAQ on how to revert to the previous behavior (2.14.0 and older, using some configuration setup) where the system-provided version is not used by default. That way people can choose to opt out of this behavior change if they want.